### PR TITLE
ci(backstage-plugin): Publish package for PRs

### PR DESCRIPTION
**Pull request by kylejwatson**
# Description

Add a publish step to the PR workflow to publish to the PR tag within the github registry.

### Issue

This PR addresses issue #99.

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/DevoteamNL/opendora/blob/main/CONTRIBUTING.md).**
- [x] **I have ran the linting and formatting scripts to make sure the code is consistently styled.**
- [x] **I have built/compiled and ran the tests to make sure the code works correctly.**
- [x] **I have added or changed tests to cover new or different functionality, where applicable.**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### CODE REVIEW

File: .github/workflows/pr-backstage-plugin-workflow.yaml [hunk 1]
```diff
@@ -46,3 +48,14 @@ jobs:
 
       - name: test
         run: yarn workspace @devoteam-nl/open-dora-backstage-plugin test:coverage
+
+      - name: publish
+        working-directory: backstage-plugin/plugins/open-dora
+        run: |
+          # Replace the @devoteam-nl with @devoteamnl in package.json
+          sed -i 's/"@devoteam-nl\//"@devoteamnl\//g' package.json
+          yarn version --new-version 0.0.$PR_NUMBER --no-git-tag-version
```
Comment (brunocribeiro) [inline .github/workflows/pr-backstage-plugin-workflow.yaml:57]: Consider using [gitversion](https://gitversion.net/) for the version as it follows SemVer and works fine for PRs as well. They also got a GH action: https://github.com/GitTools/actions

Comment (kylejwatson) [reply .github/workflows/pr-backstage-plugin-workflow.yaml]: This is a cool tool, though it's highlighted an issue with the current release setup that the version in the github registry is different from NPM registry. Probably worth keeping them in sync by publishing to both on release. For another PR anyway.

**Review by brunocribeiro — APPROVED** [review]

File: .github/workflows/pr-backstage-plugin-workflow.yaml [hunk 2]
```diff
@@ -38,11 +37,57 @@ jobs:
...
+
+      - name: Determine Version
+        id: gitversion
+        uses: gittools/actions/gitversion/execute@v0
+
+      - name: publish
+        working-directory: backstage-plugin/plugins/open-dora
+        run: |
+          # Replace the @devoteam-nl with @devoteamnl in package.json
+          sed -i 's/"@devoteam-nl\//"@devoteamnl\//g' package.json
```
Comment (goposky) [inline .github/workflows/pr-backstage-plugin-workflow.yaml:88]: What's going on here?

Comment (kylejwatson) [reply .github/workflows/pr-backstage-plugin-workflow.yaml:88]: To publish to NPM registry we need to use devoteam-nl as we have access to that org. To publish to GitHub registry we need to have the scope match the repo owner, in our case being devoteamnl

**Pull request merged**